### ISSUE TITLE
yarascan error in last REKALL version: 'float' object cannot be interpreted as integer 

### ISSUE THREAD

**Issue by labs71** [google/rekall#361]
Hello,
I'm getting an exception when trying to do a simple yara search using the last rekall from GIT. Although the installation works perfectly well, the ``yarasearch`` method is returning an error.

What I'm trying to do is:

```
testyara = 'rule testyara  { strings: $c = {C7 04 24 00 50 40 00} condition: $c }'
yarascan(pids=[3984], yara_expression=testyara, scan_process_memory=True)
```
And that's it, the process is there and it has a maching yara, but still, that's not the issue here. If there is no match I would be ok, probably, but the exception is an unexpected behavior.

I'm using python 3.6.4

Here you have the whole execution with error:

```
$ python --version
Python 3.6.4
$ rekall -f /var/tmp/memory_dump/memory.dmp


----------------------------------------------------------------------------
The Rekall Digital Forensic/Incident Response framework 1.7.3.dev10 (Hurricane Ridge).

"We can remember it for you wholesale!"

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License.

See http://www.rekall-forensic.com/docs/Manual/tutorial.html to get started.
----------------------------------------------------------------------------
[1] memory.dmp 16:33:22> testyara = 'rule testyara  { strings: $c = {C7 04 24 00 50 40 00} condition: $c }'
[1] memory.dmp 16:33:33> yarascan(pids=[3984], yara_expression=testyara, scan_process_memory=True)
2018-02-09 16:33:33,240:WARNING:rekall.1:Unable to determine file size, assuming file is volatile.
       Owner            Rule      Offset                                 hexdump                               Context
-------------------- ---------- ---------- ------------------------------------------------------------------- -------
2018-02-09 16:33:34,498:CRITICAL:rekall.1:Traceback (most recent call last):
  File "/home/user/stuff/repos/rekall/rekall-core/rekall/session.py", line 870, in RunPlugin
    result = plugin_obj.render(ui_renderer) or plugin_obj
  File "/home/user/stuff/repos/rekall/rekall-core/rekall/plugin.py", line 761, in render
    for row in self.collect():
  File "/home/user/stuff/repos/rekall/rekall-core/rekall/plugins/yarascanner.py", line 133, in collect
    for run in self.generate_memory_ranges():
  File "/home/user/stuff/repos/rekall/rekall-core/rekall/plugins/windows/common.py", line 544, in generate_memory_ranges
    regions = list(self.session.plugins.virt_map())
  File "/home/user/stuff/repos/rekall/rekall-core/rekall/plugin.py", line 379, in __iter__
    for x in self.collect():
  File "/home/user/stuff/repos/rekall/rekall-core/rekall/plugins/windows/misc.py", line 129, in collect
    key=lambda x: (int(x["virt_start"]), int(x["virt_end"]))):
  File "/home/user/stuff/repos/rekall/rekall-lib/rekall_lib/utils.py", line 1163, in __iter__
    for item in self._iterator:
  File "/home/user/stuff/repos/rekall/rekall-core/rekall/plugins/windows/misc.py", line 189, in collect_from_MiSystemVaType
    for offset in range(system_range_start, 0xffffffff, va_table_size):
TypeError: 'float' object cannot be interpreted as an integer

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-a8e9f9920768> in <module>()
----> 1 yarascan(pids=[3984], yara_expression=testyara, scan_process_memory=True)

~/stuff/repos/rekall/rekall-core/rekall/obj.py in __call__(self, *args, **kwargs)
    151         new_kwargs = self._kwargs.copy()
    152         new_kwargs.update(kwargs)
--> 153         return self._target(*(self._args + args), **new_kwargs)
    154
    155     def get_default_arguments(self):

~/stuff/repos/rekall/rekall-core/rekall/session.py in RunPlugin(self, plugin_obj, *args, **kwargs)
    871                 self.last = plugin_obj
    872             except (Exception, KeyboardInterrupt) as e:
--> 873                 self._HandleRunPluginException(ui_renderer, e)
    874
    875             finally:

~/stuff/repos/rekall/rekall-core/rekall/session.py in RunPlugin(self, plugin_obj, *args, **kwargs)
    868                     raise ValueError(
    869                         "Invalid plugin: %s" % original_plugin_obj)
--> 870                 result = plugin_obj.render(ui_renderer) or plugin_obj
    871                 self.last = plugin_obj
    872             except (Exception, KeyboardInterrupt) as e:

~/stuff/repos/rekall/rekall-core/rekall/plugin.py in render(self, renderer, **options)
    759
    760         renderer.table_header(self.table_header, **table_options)
--> 761         for row in self.collect():
    762             if isinstance(row, (list, tuple)):
    763                 renderer.table_row(*row, **options)

~/stuff/repos/rekall/rekall-core/rekall/plugins/yarascanner.py in collect(self)
    131         """Render output."""
    132         count = 0
--> 133         for run in self.generate_memory_ranges():
    134             for match, address in self.generate_hits(run):
    135                 count += 1

~/stuff/repos/rekall/rekall-core/rekall/plugins/windows/common.py in generate_memory_ranges(self)
    542         # dividing the kernel space into subregions.
    543         if not self.plugin_args.scan_kernel:
--> 544             regions = list(self.session.plugins.virt_map())
    545
    546             # Scan session pools in each process.

~/stuff/repos/rekall/rekall-core/rekall/plugin.py in __iter__(self)
    377         """
    378         if callable(getattr(self, "collect", None)):
--> 379             for x in self.collect():
    380                 if x:
    381                     yield x

~/stuff/repos/rekall/rekall-core/rekall/plugins/windows/misc.py in collect(self)
    127                     self.collect_from_MiVisibleState(),
    128                     self.collect_from_pools()),
--> 129                 key=lambda x: (int(x["virt_start"]), int(x["virt_end"]))):
    130             self.rows += 1
    131             yield x

~/stuff/repos/rekall/rekall-lib/rekall_lib/utils.py in __iter__(self)
   1161
   1162     def __iter__(self):
-> 1163         for item in self._iterator:
   1164             key = self._key(item)
   1165             if key not in self._seen:

~/stuff/repos/rekall/rekall-core/rekall/plugins/windows/misc.py in collect_from_MiSystemVaType(self)
    187         # Coalesce the ranges.
    188         range_type = range_start = range_length = 0
--> 189         for offset in range(system_range_start, 0xffffffff, va_table_size):
    190             table_index = old_div((offset - system_range_start), va_table_size)
    191             page_type = system_va_table[table_index]

TypeError: 'float' object cannot be interpreted as an integer
[1] memory.dmp 16:33:50>

```

**Comment by scudette** [comment]
Hmm this looks to be another python2 -> python3 bug. Probably the division just above produces a float instead of an int.

```                                                                                                                                                                                                                    
         # The size varies on PAE profiles.                                                                                                                                                                         
-        va_table_size = 0x1000 * 0x1000 / self.profile.get_obj_size("_MMPTE")                                                                                                                                      
+        va_table_size = 0x1000 * 0x1000 // self.profile.get_obj_size("_MMPTE")                                                                                                                                     
 ```                                                                                                                                                                                                                
 Thanks for reporting and the detailed bug report :-).

**Comment by labs71** [comment]
Thank you very much! This error is over :)

**Comment by labs71** [comment]
U didn't merge this fix to the 1.7.2rc, right? :(
